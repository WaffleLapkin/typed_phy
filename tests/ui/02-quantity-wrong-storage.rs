use typed_phy::{Quantity, units::MetrePerSecond};

fn main() {
    let _: Quantity<u32, _> = Quantity::<i32, MetrePerSecond>::new(0);
}
