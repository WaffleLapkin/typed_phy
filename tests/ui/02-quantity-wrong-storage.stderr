error[E0308]: mismatched types
 --> $DIR/02-quantity-wrong-storage.rs:4:31
  |
4 |     let _: Quantity<u32, _> = Quantity::<i32, MetrePerSecond>::new(0);
  |            ----------------   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i32`
  |            |
  |            expected due to this
  |
  = note: expected struct `typed_phy::quantity::Quantity<u32, _>`
             found struct `typed_phy::quantity::Quantity<i32, typed_phy::unit::Unit<typenum::int::PInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>>, typenum::int::Z0, typenum::int::NInt<typenum::uint::UInt<typenum::uint::UTerm, typenum::bit::B1>>, typenum::int::Z0, typenum::int::Z0, typenum::int::Z0, typenum::int::Z0>>`
